### PR TITLE
Add GBIF to our scientific names, so that our crops can associate creative commons photos

### LINKED ISSUE #3557: Link to PlantNet for crop images
https://plantnet.org/en/

Unusual crops don't have photos.
Plantnet has an abundance.

Can we link to them?
IE: https://identify.plantnet.org/k-australia/species/Capsicum%20annuum%20L./data

### PR DESCRIPTION
**Pull request by CloCkWeRX**
Fixes #3557
Partially improves #1919

# Manual testing done so far

- [x] Rake task against crops in staging
- [x] Manually fetch data
- [ ] Manually add / edit a scientific name - executed, but didn't see the expected results
- [x] Photo show

# Autocompletes when wrangling

Add an autocomplete control for editing a scientific name.
Couldn't (easily) the one added by the +/- controls in.

When you lookup a scientific name, you get suggestions:
image

Behind the scenes:
image

# Importing photos

Before:
image

After:
https://growstuff-staging.herokuapp.com/crops/rice-bean
image

Which has all of the right links:
image
https://www.inaturalist.org/photos/317727679

# TODO

- [x] Add to crop/show page with links if the GBIF id is available for the scientific name
- [x] Add a non autocomplete version that looks up photos
- [ ] Add a version of this to allow a wikidata lookup
- [ ] Have the alternative names suggest from GBIF
- [x] Add a rake task or similar that allows you to "find crops without photos but with a GBIF id, and look for Pl@ntNet images"
- [x] Get rid of accessibilitylint. Did I need 500 repeated notifications on unchanged code?
- [x] Get rid of codefactor. I don't care if recorded spec cassettes have lines too long.
- [ ] Cater for match not found behaviours to avoid spamming the upstream API?

Manual step:

Make sure we run 
`bundle exec rake db:migrate`
then
`bundle exec rake gbif:import`
 on heroku deploy.

Not sure if we want to set up a monthly cron or similar

### CODE REVIEW

File: app/views/crops/_form.html.haml [hunk 1]
```diff
@@ -65,8 +65,12 @@
             .col-2
               = label_tag :scientific_names, "Scientific name #{index + 1}:", class: 'control-label'
             .col-8
-              = text_field_tag "sci_name[#{index + 1}]", sci.name, id: "sci_name[#{index + 1}]", class: 'form-control'
-              %span.help-block Scientific name of crop.
+              = text_field_tag "sci_name[#{index + 1}]", sci.name, id: "sci_name[#{index + 1}]",
```
Comment (CloCkWeRX) [inline app/views/crops/_form.html.haml]:
```suggestion
              = text_field_tag "sci_name[#{index + 1}]", sci.name, id: "sci_name[#{index + 1}]",
```

File: app/views/crops/_form.html.haml [hunk 2]
```diff
@@ -65,8 +65,12 @@
             .col-2
               = label_tag :scientific_names, "Scientific name #{index + 1}:", class: 'control-label'
             .col-8
-              = text_field_tag "sci_name[#{index + 1}]", sci.name, id: "sci_name[#{index + 1}]", class: 'form-control'
-              %span.help-block Scientific name of crop.
+              = text_field_tag "sci_name[#{index + 1}]", sci.name, id: "sci_name[#{index + 1}]", 
+                                                                   class: 'scientific-name-auto-suggest form-control',
```
Comment (CloCkWeRX) [inline app/views/crops/_form.html.haml]:
```suggestion
                                                                    class: 'scientific-name-auto-suggest form-control',
```

File: app/views/crops/_form.html.haml [hunk 3]
```diff
@@ -65,8 +65,12 @@
...
+                                                                   data: { source_url: gbif_suggest_scientific_names_path }
```
Comment (CloCkWeRX) [inline app/views/crops/_form.html.haml]:
```suggestion
                                                                    data: { source_url: gbif_suggest_scientific_names_path }
```

File: app/views/crops/_form.html.haml [hunk 4]
```diff
@@ -65,8 +65,12 @@
...
+              %span.help-block Searches GBIF to determine scientific name of crop.
+              = hidden_field_tag "sci_gbif_key[#{index + 1}]", sci.gbif_key, id: "sci_gbif_key[#{index + 1}]",
+                                                                             class: 'scientific-name-auto-suggest-id'
```
Comment (CloCkWeRX) [inline app/views/crops/_form.html.haml]:
```suggestion
                                                                              class: 'scientific-name-auto-suggest-id'
```

**Comment by CloCkWeRX** [comment]
Well, I think that's "ready enough".

It's a bit annoying because you can't type to find a scientific name, then add another, and repeat.
But otherwise, it should let you manually refresh, and sync data.

File: Gemfile.lock [hunk 5]
```diff
@@ -745,4 +754,4 @@ RUBY VERSION
     ruby 3.1.4p223
 
 BUNDLED WITH
-   2.3.11
+    2.3.11
```
Comment (CloCkWeRX) [inline Gemfile.lock]:
```suggestion
   2.3.11
```

**Comment by CloCkWeRX** [comment]
https://growstuff-staging.herokuapp.com/crops/bok-choy working nicely!

File: lib/tasks/gbif.rake [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+namespace :gbif do
+  desc "Retrieve crop info from open farm"
```
Comment (CloCkWeRX) [inline lib/tasks/gbif.rake]:
```suggestion
  desc "Retrieve crop info from GBIF"
```

File: app/views/crops/_scientific_names.html.haml [hunk 7]
```diff
@@ -16,7 +16,13 @@
               = delete_icon
               = t('.delete')
       - else
-        .badge= sn.name
+        - if sn.gbif_key = 3084850
```
Comment (CloCkWeRX) [inline app/views/crops/_scientific_names.html.haml]:
```suggestion
        - if sn.gbif_key
```

**Comment by CloCkWeRX** [comment]
I marked all of the codeclimate line lengths things as a WONTFIX.
Recommend doing this as a squash merge to keep history clean.

**Comment by cesy** [comment]
Great idea, looks good in principle. Monthly cron sounds like a good idea to keep it updated.

File: app/views/photos/new.html.haml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 - content_for :title, "New Photo"
 
 %h1 New Photo
-%h2 Choose photo for #{link_to @item, @item}
+%h2 Choose photo for #{link_to @item, @item} from Flickr, or contribute to unique crops to <a href="https://inaturalist.org/" target="_blank">iNaturalist</a> or <a href="https://identify.plantnet.org/" target="_blank">Pl@antNet</a> via the app.
```
Comment (cesy) [inline app/views/photos/new.html.haml]: Extra a in Pl@antNet?

File: app/views/plantings/_photos.haml [hunk 9]
```diff
@@ -10,5 +10,5 @@
 - else
   %p No photos.
   - if can?(:edit, planting) && can?(:create, Photo)
-    %p Add a photo to visually track growth of this planting
+    %p Add a photo to visually track growth of this planting, to Flickr, iNaturalist or Pl@antNet
```
Comment (cesy) [inline app/views/plantings/_photos.haml]: Extra a in Pl@ntNet comparing to their name on their site

Comment (CloCkWeRX) [inline app/views/plantings/_photos.haml]:
```suggestion
    %p Add a photo to visually track growth of this planting, to Flickr, iNaturalist or Pl@ntNet
```

File: app/views/photos/new.html.haml (hunk 8 above)
Comment (CloCkWeRX) [inline app/views/photos/new.html.haml]:
```suggestion
%h2 Choose photo for #{link_to @item, @item} from Flickr, or contribute to unique crops to <a href="https://inaturalist.org/" target="_blank">iNaturalist</a> or <a href="https://identify.plantnet.org/" target="_blank">Pl@ntNet</a> via the app.
```

File: lib/tasks/gbif.rake [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+namespace :gbif do
+  desc "Retrieve crop info from GBIF"
+  # usage: rake growstuff:admin_user name=skud
```
Comment (CloCkWeRX) [inline lib/tasks/gbif.rake]:
```suggestion
```

**Comment by CloCkWeRX** [comment]
I... am about to head to sleep, but if you have a bit of time to deploy to staging, that might be well worth considering!

**Comment by CloCkWeRX** [comment]
Okay, going to merge this one once the build is green. The monthly scheduler/cron job, will have to think about at time of deploy.

**Pull request merged**